README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# gsloid: global sea level and oxygen isotope data

[![Travis-CI Build
Status](https://travis-ci.org/benmarwick/gsloid.png?branch=master)](https://travis-ci.org/benmarwick/gsloid)
[![CRAN_Status_Badge](http://www.r-pkg.org/badges/version/gsloid)](https://cran.r-project.org/package=gsloid)
![CRAN_Downloads_Badge](http://cranlogs.r-pkg.org/badges/gsloid)

The goal of gsloid is to make available raw data for sea level curves
and
![\delta](https://latex.codecogs.com/png.image?%5Cdpi%7B110%7D&space;%5Cbg_white&space;%5Cdelta "\delta")<sup>18</sup>O
curves for the Holocene and most of the Pleistocene.

## Installation

You can install gsloid from CRAN with:

``` r
inistall.pacakges("gsloid")
```

Or you can install the development version of gsloid from github with:

``` r
# install.packages("devtools")
devtools::install_github("benmarwick/gsloid")
```

## Overview

This package includes commonly used datasets in palaeoecology and
archaeology:

-   Global sea level curve
-   Global oxygen isotope curve
-   Boundary ages for Marine Isotope Stages

There are many possible sources for these kinds of data, this package
includes data from:

| Dataset      | Source                                                                                                                                                                                       |
|:-------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| lisiecki2005 | Lisiecki, L.E. and M.E. Raymo. 2005. A Pliocene-Pleistocene stack of 57 globally distributed benthic D18O records. Paleoceanography, Vol. 20, PA1003, <https://doi.org/10.1029/2004PA001071> |
| spratt2016   | Spratt, Rachel M. and Lorraine E. Lisiecki 2016. A Late Pleistocene sea level stack. Climate of the Past. Vol. 12, 1079-1092, <https://doi.org/10.5194/cp-12-1-2016>                         |

You **must** cite those sources in the table above if you use the data
in this package.

Here’s the structure of the main data sets:

``` r
library(gsloid)
str(lisiecki2005)
#> 'data.frame':    2115 obs. of  3 variables:
#>  $ Time : num  0 1 2 3 4 5 6 7 8 9 ...
#>  $ d18O : num  3.23 3.23 3.18 3.29 3.3 3.26 3.33 3.37 3.42 3.38 ...
#>  $ Error: num  0.03 0.04 0.03 0.03 0.03 0.03 0.04 0.04 0.03 0.04 ...
```

``` r
str(spratt2016)
#> 'data.frame':    799 obs. of  9 variables:
#>  $ age_calkaBP            : num  0 1 2 3 4 5 6 7 8 9 ...
#>  $ SeaLev_shortPC1        : num  8.49 7.63 4.01 4.35 3.13 ...
#>  $ SeaLev_shortPC1_err_sig: num  5.23 4.87 4.83 4.72 4.74 4.57 5.04 5.9 6.79 8.3 ...
#>  $ SeaLev_shortPC1_err_lo : num  -1.72 -2.9 -4.51 -6.93 -10.43 ...
#>  $ SeaLev_shortPC1_err_up : num  17.93 16.39 13.59 12.08 8.41 ...
#>  $ SeaLev_longPC1         : num  8.96 7.72 5.96 3.54 1.88 0 -2 -5.38 -7.12 -11.6 ...
#>  $ SeaLev_longPC1_err_sig : num  5.72 5.13 4.69 4.42 4.39 ...
#>  $ SeaLev_longPC1_err_lo  : num  -1.21 -2.77 -5.01 -7.28 -10.54 ...
#>  $ SeaLev_longPC1_err_up  : num  20.38 17.1 14.21 10.9 7.63 ...
```

``` r
str(LR04_MISboundaries)
#> 'data.frame':    232 obs. of  7 variables:
#>  $ MIS_Boundary     : chr  "1/2" "2/3" "3/4" "4/5" ...
#>  $ start_MIS        : chr  "1" "2" "3" "4" ...
#>  $ end_MIS          : chr  "2" "3" "4" "5" ...
#>  $ label_MIS        : chr  "1" "2" "3" "4" ...
#>  $ LR04_Age_ka_start: num  14 29 57 71 82 87 96 109 123 130 ...
#>  $ LR04_Age_ka_end  : num  0 14 29 57 71 82 87 96 109 123 ...
#>  $ LR04_Age_ka_mid  : num  7 21.5 43 64 76.5 ...
```

Detailed descriptions of the variables are avaliable in the data
documentation, run `?lisiecki2005`, `?spratt2016` and
`?LR04_MISboundaries` for more information.

## Usage

Atlhough these data are suitable for many kinds of analyses, the primary
reason that I made this package is so I can easily make my own plots of
these data. Here’s how I typically start with plotting the oxygen
isotope data. I have set limits on the x-axis so it only shows 0-250 ka
because that’s what I’m interested in:

``` r
library(ggplot2)


ggplot(lisiecki2005, 
       aes(Time,
           d18O)) +
  geom_line() +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()
```

![](vignettes/figures/README-unnamed-chunk-6-1.png)<!-- -->

And here’s how I start to plot the sea level data:

``` r
ggplot(spratt2016, 
       aes(age_calkaBP,
           SeaLev_shortPC1)) +
  geom_line() +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_continuous(name = "Sea Level, meters above present day") +
  theme_bw()
```

![](vignettes/figures/README-unnamed-chunk-7-1.png)<!-- -->

Often we want to see the Marine istope stages on these plots also. This
package includes the dataset `LR04_MISboundaries` of start and end dates
for each stage, so we can draw these stages easily. I obtained these
data from Lorraine Lisiecki’s web page:
<http://lorraine-lisiecki.com/LR04_MISboundaries.txt>

Some care is required to get the MIS numbers positions in easy-to-read
locations. Here I use `rep()` and `seq()` to help position the numbers.

``` r
# subset the MIS data for the last 250 ka years
mis_last_250ka <- LR04_MISboundaries[LR04_MISboundaries$LR04_Age_ka_start <= 250, ]

# plot the oxygen istope data line on top of the MIS lines
ggplot() +
  geom_vline(data = mis_last_250ka,          # add MIS lines
             aes(xintercept = LR04_Age_ka_start),
             colour = "blue") +
  annotate("text", 
           label = mis_last_250ka$label_MIS, 
           x = mis_last_250ka$LR04_Age_ka_mid,
           y = c(rep(3.0, 4), 
                 seq(3.2, 2.7, length.out = 6), 
                 rep(3.0, 2)),
           size = 3) +
  geom_line(data = lisiecki2005,                # add d18O line
            aes(Time,
                d18O)) +  
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()
```

![](vignettes/figures/README-unnamed-chunk-8-1.png)<!-- -->

Sometimes we prefer to indicate the MIS by horizontal lines. Once again
we have to carefully place the line segments and labels so they are
clear to read:

``` r
ggplot() +
  geom_segment(data = mis_last_250ka, # add MIS lines
               aes(x =    LR04_Age_ka_end,
                   xend = LR04_Age_ka_start,
                   y =    seq(3, 2.5, length.out = nrow(mis_last_250ka)),
                   yend = seq(3, 2.5, length.out = nrow(mis_last_250ka))),
               colour = "blue",
               size = 1) +
  annotate("text", 
           label = mis_last_250ka$label_MIS, 
           x =     mis_last_250ka$LR04_Age_ka_mid,
           y = c(seq(2.9, 2.7, length.out = 4), 
                 seq(3.1, 2.8, length.out = 5),
                 seq(2.4, 2.3, length.out = 3)),
           size = 3) +
  geom_line(data = lisiecki2005,                # add d18O line
            aes(Time,
                d18O)) +  
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()
```

![](vignettes/figures/README-unnamed-chunk-9-1.png)<!-- -->

And sometimes we might want shaded rectangles to indicate the MIS:

``` r
ggplot() +
  annotate("rect", 
           xmin = mis_last_250ka$LR04_Age_ka_end, 
           xmax = mis_last_250ka$LR04_Age_ka_start, 
           ymin = -Inf, 
           ymax = Inf,
        alpha = .2,
        fill = rep(c("grey70", "white"), 
                   nrow(mis_last_250ka)/2)) +
  annotate("text", 
           label = mis_last_250ka$label_MIS, 
           x =     mis_last_250ka$LR04_Age_ka_mid,
           y = c(rep(3, 4), 
                 seq(3.1, 2.8, length.out = 5),
                 rep(3, 3)),
           size = 3) +
  geom_line(data = lisiecki2005,                # add d18O line
            aes(Time,
                d18O)) +  
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()
```

![](vignettes/figures/README-unnamed-chunk-10-1.png)<!-- -->

Maybe we want the MIS regions by themselves so we can plot some other
time series besides the ones includes here. We need to adjust the `y`
values in the `annotate()` function to ensure that the MIS labels show
in a readble way. Here I show an imaginary variable that might have a
maximum value of about 10. So I adjust the `y` values in the
`annotate()` function to position the MIS labels just below 10 on the y
axis.

``` r
ggplot() +
  annotate("rect", 
           xmin = mis_last_250ka$LR04_Age_ka_end , 
           xmax = mis_last_250ka$LR04_Age_ka_start, 
           ymin = -Inf, 
           ymax = Inf,
        alpha = .2,
        fill = rep(c("grey70", "white"), 
                   nrow(mis_last_250ka)/2)) +
  annotate("text", 
           label = mis_last_250ka$label_MIS, 
           x =     mis_last_250ka$LR04_Age_ka_mid, 
           y = c(rep(9, 4), 
                 seq(8.1, 9.9, length.out = 5),
                 rep(9, 3)),
           size = 3) +
  # add in other geoms here...
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_continuous(limits = c(0, 10),
                  name = "Some other variable (units)") +
  theme_bw()
```

![](vignettes/figures/README-unnamed-chunk-11-1.png)<!-- -->

Please note that this project is released with a [Contributor Code of
Conduct](https://github.com/benmarwick/gsloid/blob/master/CONDUCT.md).
By participating in this project you agree to abide by its terms.


NEWS.md
# gsloid 0.2 (2022-05-08)
=========================

* Responded to an email from CRAN about a NOTE on the CONDUCT.md 

# gsloid 0.2 (2022-05-03)
=========================

* Responded to an email from CRAN about a NOTE on the vignette title. 
* Minor edits and spell fixes to documentation 

# gsloid 0.1.9000 (2017-06-30)
=========================

Now on CRAN

# gsloid 0.1.9000 (2017-06-27)
=========================

* Added a `NEWS.md` file to track changes to the package.





revdep/checks.noindex/deeptime/old/deeptime.Rcheck/tests/testthat.R
library(testthat)
library(deeptime)

test_check("deeptime")


revdep/checks.noindex/deeptime/old/deeptime.Rcheck/tests/testthat/setup-data.R
suppressPackageStartupMessages(library(divDyn, quietly = TRUE))
suppressPackageStartupMessages(library(gsloid, quietly = TRUE))
suppressPackageStartupMessages(library(tidyverse, quietly = TRUE))
data(corals)
suppressPackageStartupMessages(library(paleotree, quietly = TRUE))
data(RaiaCopesRule)

coral_div <- corals %>% filter(stage != "") %>%
  group_by(stage) %>%
  summarise(n = n()) %>%
  mutate(stage_age = (stages$max_age[match(stage, stages$name)] + stages$min_age[match(stage, stages$name)])/2)

coral_div_diet <- corals %>% filter(stage != "") %>%
  group_by(diet, stage) %>%
  summarise(n = n()) %>%
  mutate(stage_age = (stages$max_age[match(stage, stages$name)] + stages$min_age[match(stage, stages$name)])/2)

coral_div_dis <- corals %>% filter(period != "") %>%
  group_by(diet, period) %>%
  summarise(n = n()) %>%
  mutate(period_age = (periods$max_age[match(period, periods$name)] + periods$min_age[match(period, periods$name)])/2) %>%
  arrange(-period_age)


revdep/checks.noindex/deeptime/new/deeptime.Rcheck/tests/testthat/helper-vdiffr.R
# Copied from ggplot2

# By default, if vdiffr is not installed, all visual tests are skipped unless
# VDIFFR_RUN_TESTS is explicitly set to "true", which should be the case only on
# a GitHub Actions CI runner with stable version of R.

if (requireNamespace("vdiffr", quietly = TRUE) && utils::packageVersion('testthat') >= '3.0.3') {
  expect_doppelganger <- vdiffr::expect_doppelganger
} else {
  # If vdiffr is not available and visual tests are explicitly required, raise error.
  if (identical(Sys.getenv("VDIFFR_RUN_TESTS"), "true")) {
    abort("vdiffr is not installed")
  }

  # Otherwise, assign a dummy function
  expect_doppelganger <- function(...) skip("vdiffr is not installed.")
}


R/data.R
#' LR04 Global Pliocene-Pleistocene Benthic d18O Stack (5.3-Myr).
#'
#' The LR04 stack spans 5.3 Myr and is an average of 57 globally distributed benthic d18O records (which measure global ice volume and deep ocean temperature) collected from the scientific literature. Obtained from ftp://ftp.ncdc.noaa.gov/pub/data/paleo/contributions_by_author/lisiecki2005/lisiecki2005.txt on 28 June 2017
#'
#' NAME OF DATA SET:
#' LR04 Global Pliocene-Pleistocene Benthic d18O Stack
#' LAST UPDATES: 8/2005 (Change number of significant digits in LR04 stack)
#' 7/2005 (final version, 0.64 permil correction for Cibicidoides added)
#' CONTRIBUTOR: Lorraine E. Lisiecki, Brown University
#' IGBP PAGES/WDCA CONTRIBUTION SERIES NUMBER: 2005-008
#'
#'
#' SUGGESTED DATA CITATION:
#' Lisiecki, L.E. and M.E. Raymo.  2005.
#' LR04 Global Pliocene-Pleistocene Benthic d18O Stack.
#' IGBP PAGES/World Data Center for Paleoclimatology
#' Data Contribution Series #2005-008.
#' NOAA/NGDC Paleoclimatology Program, Boulder CO, USA.
#'
#'
#' ORIGINAL REFERENCE:
#' Lisiecki, L.E. and M.E. Raymo. 2005.
#' A Pliocene-Pleistocene stack of 57 globally distributed benthic D18O records.
#' Paleoceanography, Vol. 20, PA1003, doi:10.1029/2004PA001071.
#'
#' ADDITIONAL REFERENCES:
#'
#'   Imbrie, J., J. D. Hays, D. G. Martinson, A. McIntyre, A. C. Mix, J. J. Morley,
#' N. G. Pisias, W. L. Prell, and N. J. Shackleton. 1984.
#' The orbital theory of Pleistocene climate: Support from a revised chronology
#' of the marine d18O record. in Milankovitch and Climate, Part 1, edited by
#' A. Berger, pp. 269-305, Springer, New York.
#'
#' Shackleton, N. J. 1995. New data on the evolution of Pliocene climate variability.
#' in Paleoclimate and Evolution, With Emphasis on Human Origins,
#' edited by E. S. Vrba et al.,  pp. 242-248, Yale Univ. Press, New Haven, Ct.
#'
#' ABSTRACT (Lisiecki and Raymo 2005):
#'   We present a 5.3-Myr stack (the "LR04" stack) of benthic d18O records
#' from 57 globally distributed sites aligned by an automated graphic
#' correlation algorithm. This is the first benthic d18O stack composed of more
#' than three records to extend beyond 850 ka, and we use its improved signal
#' quality to identify 24 new marine isotope stages in the early Pliocene.
#' We also present a new LR04 age model for the Pliocene-Pleistocene derived
#' from tuning the d18O stack to a simple ice model based on 21 June insolation
#' at 65N. Stacked sedimentation rates provide additional age model constraints
#' to prevent overtuning. Despite a conservative tuning strategy, the LR04 benthic
#' stack exhibits significant coherency with insolation in the obliquity band
#' throughout the entire 5.3 Myr and in the precession band for more than half
#' of the record. The LR04 stack contains significantly more variance in benthic
#' d18O than previously published stacks of the late Pleistocene as the result
#' of higher resolution records, a better alignment technique, and a greater
#' percentage of records from the Atlantic. Finally, the relative phases of the
#' stack’s 41- and 23-kyr components suggest that the precession component of
#' d18O from 2.7–1.6 Ma is primarily a deep-water temperature signal and that
#' the phase of d18O precession response changed suddenly at 1.6 Ma.
#'
#'
#'
#' PERIOD OF RECORD:  5.3 MMYrBP - present
#'
#' GEOGRAPHIC REGION: Global
#'
#' DESCRIPTION:
#'   Lisiecki-Raymo 2004 (LR04) Global Pliocene-Pleistocene Benthic D18O Stack.
#'
#' The LR04 stack spans 5.3 Myr and was constructed by graphically aligning
#' 57 globally distributed benthic d18O records. Its age model is orbitally tuned
#' to June 21 insolation at 65N but is also constrained by the average sedimentation
#' rates of the 57 sites.
#'
#' LR04 Global Pliocene-Pleistocene Benthic d18O Stack
#
#'
#' @format A data frame with 2115 rows and 3 variables:
#' \describe{
#'   \item{Time}{x 1000 years (i.e. ka)}
#'   \item{d18O}{Benthic d18O (per mil)}
#'   \item{Error}{Standard error (per mil)}
#'   ...
#' }
#' @source \url{ftp://ftp.ncdc.noaa.gov/pub/data/paleo/contributions_by_author/lisiecki2005/lisiecki2005.txt}
#' @examples
#' names(lisiecki2005)
#' head(lisiecki2005)
#' # plot for 0-250 ka:
#' if (require("ggplot2")) {
#'    ggplot(lisiecki2005,
#'           aes(Time,
#'               d18O)) +
#'      geom_line() +
#'      scale_x_continuous(limits = c(0, 250),
#'                         name = "x 1000 years ago") +
#'      scale_y_reverse(name = bquote(delta^18*O)) +
#'      theme_bw()
#'    }
"lisiecki2005"


#'
#'
#' Global Sea Level Reconstruction using Stacked Records from 0-800 ka.
#'
#' This is a Late Pleistocene sea level stack based on marine sediment core data (foraminiferal carbonate d18O) as estimated by several different techniques in seven different studies. Obtained from https://www.ncdc.noaa.gov/paleo-search/study/19982 on 28 June 2017
#'
#' # Online_Resource: http://www.ncdc.noaa.gov/paleo/study/19982
#'  Online_Resource: http://www1.ncdc.noaa.gov/pub/data/paleo/paleocean/by_contributor/spratt2016/spratt2016.txt
#'
#'  Archive: Paleoceanography
#'
#'
#'   Contribution_Date
#'     Date: 2016-04-12
#'
#'  Title
#'     Study_Name: Global Sea Level Reconstruction using Stacked Records from 0-800 ka
#'
#'
#'  Investigators
#'       Investigators: Spratt, Rachel M.; Lisiecki, Lorraine E.
#'
#'
#'  Description and Notes
#'         Description: Late Pleistocene sea level stack based on marine sediment core data (foraminiferal carbonate d18O) as estimated by several different techniques in seven different studies. Sea level stack is a compilation data from the following published sea level reconstructions: Bintanja, R., Roderik, S.W., van de Wal, O. J.: Modeled atmospheric temperatures and global sea levels over the past million years, Nature, 437, 125-128  doi:10.1038/nature03975, 2005. ; Elderfield, H., Ferretti, P., Greaves, M., Crowhurst, S.J., McCave, I.N., Hodell, D.A., Piotrowski, A.M.: Evolution of ocean temperature and ice volume through the Mid-Pleistocene Climate Transition, Science, 337,6095, 704-709, doi:10.1126/science.1221294, 2012.; Rohling, E.J., Grant, K., Bolshaw, M., Roberts, A.P., Siddall, M., Hemleben, C., Kucera, M.: Antarctic temperature and global sea level closely coupled over the past five glacial cycles. Nat. Geosci., 2, 500-504, 2009.; Rohling, E.J., Grant, K.M., Bolshaw, M., Roberts, A. P., Siddall, M., Hemleben C. Kucera, M., Foster, G.L., Marino, G., Roberts, A.P., Tamisiea, M.E., and Williams, F.:  Sea-level and deep-sea-temperature variability over the past 5.3 million years, Nature, 508, 477-482, 2014.; Shakun, J.D., Lea, D.W., Lisiecki, L.E., Raymo, M.E.: An 800-kyr record of global surface ocean d18O and implications for ice volume-temperature coupling, Earth. Planet. Sc. Lett., 426, 58-68, 2015.; Sosdian S., Rosenthal Y.: Deep-Sea Temperature and Ice Volume Changes Across the Pliocene-Pleistocene Climate Transitions, Science, 17, 325, 5938, 306-310, doi: 10.1126/science.1169938, 2009.; Waelbroeck, C., Labeyrie, L.,  Michel, E., Duplessy J.C., McManus J.: Sea-level and deep water temperature changes derived from benthic foraminifera isotopic records, Quaternary Sci. Rev., 21, 295-305, 2002.
#'
#'
#'  Publication
#'  Authors: Spratt, Rachel M. and Lorraine E. Lisiecki
#'  Published_Date_or_Year: 2016
#'  Published_Title: A Late Pleistocene sea level stack
#'  Journal_Name: Climate of the Past
#'  Volume: in press
#'  Edition:
#'  Issue:
#'  Pages:
#'  Report Number:
#'  DOI: 10.5194/cp-12-1-2016
#'
#'
#'  Abstract: Late Pleistocene sea level has been reconstructed from ocean sediment core data using a wide variety of proxies and models. However, the accuracy of individual reconstructions is limited by measurement error, local variations in salinity and temperature, and assumptions particular to each technique. Here we present a sea level stack (average) which increases the signal-to-noise ratio of individual reconstructions. Specifically, we perform principal component analysis (PCA) on seven records from 0-430 ka and five records from 0-798 ka. The first principal component, which we use as the stack, describes ~80% of the variance in the data and is similar using either five or seven records. After scaling the stack based on Holocene and Last Glacial Maximum (LGM) sea level estimates, the stack agrees to within 5 m with isostatically adjusted coral sea level estimates for Marine Isotope Stages 5e and 11 (125 and 400 ka, respectively). Bootstrapping and random sampling yield mean uncertainty estimates of 9-12 m (1s) for the scaled stack. Sea level change accounts for about 45% of the total orbital-band variance in benthic d18O, compared to a 65% contribution during the LGM-to-Holocene transition. Additionally, the second and third principal components of our analyses reflect differences between proxy records associated with spatial variations in the d18O of seawater.
#'
#'
#'  Site Information
#'  Site_Name: Global 61.5N to 41S
#'  Location:
#'  Country:
#'  Northernmost_Latitude: 61.425
#'  Southernmost_Latitude: -40.937
#'  Easternmost_Longitude: 180
#'  Westernmost_Longitude: -180
#'
#'
#'  Data_Collection
#'    Collection_Name: global_sealevel_stack Spratt16
#'    First_Year: -798000
#'    Last_Year: 0
#'    Time_Unit: cal yr BP
#'    Core_Length:
#'    Notes: Sea level stack is a compilation data from the following published sea level reconstructions: Bintanja, R., Roderik, S.W., van de Wal, O. J.: Modeled atmospheric temperatures and global sea levels over the past million years, Nature, 437, 125-128  doi:10.1038/nature03975, 2005. ; Elderfield, H., Ferretti, P., Greaves, M., Crowhurst, S. J., McCave, I.N., Hodell, D. A., Piotrowski, A. M.: Evolution of ocean temperature and ice volume through the Mid-Pleistocene Climate Transition, Science, 337,6095, 704-709, doi:10.1126/science.1221294, 2012.; Rohling, E.J., Grant, K., Bolshaw, M., Roberts, A.P., Siddall, M., Hemleben, C., Kucera, M.: Antarctic temperature and global sea level closely coupled over the past five glacial cycles. Nat. Geosci., 2, 500–504, 2009.; Rohling, E. J., Grant, K.M., Bolshaw, M., Roberts, A. P., Siddall, M., Hemleben C. Kucera, M., Foster, G.L., Marino, G., Roberts, A.P., Tamisiea, M.E., and Williams, F.:  Sea-level and deep-sea-temperature variability over the past 5.3 million years, Nature, 508, 477–482, 2014.; Shakun, J. D., Lea, D. W., Lisiecki, L. E., Raymo, M. E.: An 800-kyr record of global surface ocean d18O and implications for ice volume-temperature coupling, Earth. Planet. Sc. Lett., 426, 58–68, 2015.; Sosdian S., Rosenthal Y.: Deep-Sea Temperature and Ice Volume Changes Across the Pliocene-Pleistocene Climate Transitions, Science, 17, 325, 5938, 306-310, doi: 10.1126/science.1169938, 2009.; Waelbroeck, C., Labeyrie, L.,  Michel, E., Duplessy J.C., McManus J.: Sea-level and deep water temperature changes derived from benthic foraminifera isotopic records, Quaternary Sci. Rev., 21, 295–305, 2002.
#'
#'
#'
#'  Chronology_Information
#'  Chronology:
#'  For cores with benthic foraminiferal carbonate d18O, age models were developed by aligning the carbonate d18O to the "LR04" benthic d18O stack (Lisiecki and Raymo, Paleoceanography, 2005). In most cases, this was the published timescale of the original record. For cores which lacked benthic carbonate d18O, their sea level estimates were aligned to the sea level estimates of the records that were already on the LR04 age model.
#'  #'  Missing Value: NaN
#'
#'  Variables
#'  Data variable description include: what, material, error, units, seasonality, archive, detail, method, C or N for Character or Numeric data)
#'
#' #@format A data frame with 799 rows and 9 variables:
#' \describe{
#'   \item{age_calkaBP}{Age, calendar ka BP}
#'   \item{SeaLev_shortPC1}{Sea Level, meters above present day, climate reconstructions, Scaled first principal component of seven sea level reconstructions (0-430 ka),N}
#'   \item{SeaLev_shortPC1_err_sig}{Sea Level, standard deviation from bootstrap, meters,,climate reconstructions, Scaled first principal component of seven sea level reconstructions (0-430 ka),N}
#'   \item{SeaLev_shortPC1_err_lo}{Sea Level, 95% confidence interval, lower bound, meters,,climate reconstructions, Scaled first principal component of seven sea level reconstructions (0-430 ka),N}
#'   \item{SeaLev_shortPC1_err_up}{Sea Level, 95% confidence interval, upper bound, meters,,climate reconstructions, Scaled first principal component of seven sea level reconstructions (0-430 ka),N}
#'   \item{SeaLev_longPC1}{Sea Level, meters above present day, climate reconstructions, Scaled first principal component of five sea level reconstructions (0-798 ka),N}
#'   \item{SeaLev_longPC1_err_sig}{Sea Level, standard deviation from bootstrap,meters,,climate reconstructions, Scaled first principal component of five sea level reconstructions (0-798 ka),N}
#'   \item{SeaLev_longPC1_err_lo}{Sea Level, 95% confidence interval, lower bound,meters,,climate reconstructions,Scaled first principal component of five sea level reconstructions (0-798 ka),N}
#'   \item{SeaLev_longPC1_err_up}{Sea Level, 95% confidence interval, upper bound,meters,,climate reconstructions, Scaled first principal component of five sea level reconstructions (0-798 ka),N}
#'   ...
#' }
#' @source \url{https://www.ncdc.noaa.gov/paleo-search/study/19982}
#'
#' @examples
#' names(spratt2016)
#' head(spratt2016)
#'
#'
"spratt2016"


#' Marine isotope stages (MIS) boundaries.
#'
#' From http://www.lorraine-lisiecki.com/LR04_MISboundaries.txt
#'
#'
#' @format A data frame with 232 rows and 7 variables:
#' \describe{
#'   \item{MIS_Boundary}{Marine isotope stage boundary, start/end}
#'   \item{start_MIS}{start of this phase}
#'   \item{end_MIS}{end of this phase}
#'   \item{label_MIS}{short version of 'start_MIS' suitable for annotating plots}
#'   \item{LR04_Age_ka_start}{Age of start of MIS, x 1000 years ago}
#'   \item{LR04_Age_ka_end}{Age of end of MIS, x 1000 years ago}
#'   \item{LR04_Age_ka_mid}{Age of middle of MIS, x 1000 years ago, suitable for controlling label placement on plots}
#' }
#' @source \url{http://www.lorraine-lisiecki.com/LR04_MISboundaries.txt}
#'
#' @examples
#' names(LR04_MISboundaries)
#' head(LR04_MISboundaries)
#' # subset the MIS data for the last 250 ka years
#' mis_last_250ka <- LR04_MISboundaries[LR04_MISboundaries$LR04_Age_ka_start <= 250, ]
"LR04_MISboundaries"



vignettes/Introduction.R
## ----echo = FALSE-------------------------------------------------------------
knitr::opts_chunk$set( fig.width =  183 * 0.0393701,  # nature's two col width
                       message = FALSE,
                       warning = FALSE)

## ---- echo = FALSE------------------------------------------------------------
references <- data.frame(Dataset = c("lisiecki2005", "spratt2016"),
                         Source = c("Lisiecki, L.E. and M.E. Raymo. 2005. A Pliocene-Pleistocene stack of 57 globally distributed benthic D18O records. Paleoceanography, Vol. 20, PA1003, <https://doi.org/10.1029/2004PA001071>",
                                    "Spratt, Rachel M. and Lorraine E. Lisiecki 2016. A Late Pleistocene sea level stack. Climate of the Past. Vol. 12, 1079-1092, <https://doi.org/10.5194/cp-12-1-2016>"))

knitr::kable(references)

## -----------------------------------------------------------------------------
library(gsloid)
str(lisiecki2005)

## -----------------------------------------------------------------------------
str(spratt2016)

## -----------------------------------------------------------------------------
str(LR04_MISboundaries)

## -----------------------------------------------------------------------------
library(ggplot2)


ggplot(lisiecki2005,
       aes(Time,
           d18O)) +
  geom_line() +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()

## -----------------------------------------------------------------------------
ggplot(spratt2016,
       aes(age_calkaBP,
           SeaLev_shortPC1)) +
  geom_line() +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_continuous(name = "Sea Level, meters above present day") +
  theme_bw()

## -----------------------------------------------------------------------------
# subset the MIS data for the last 250 ka years
mis_last_250ka <- LR04_MISboundaries[LR04_MISboundaries$LR04_Age_ka_start <= 250, ]

# plot the oxygen istope data line on top of the MIS lines
ggplot() +
  geom_vline(data = mis_last_250ka,          # add MIS lines
             aes(xintercept = LR04_Age_ka_start),
             colour = "blue") +
  annotate("text",
           label = mis_last_250ka$label_MIS,
           x = mis_last_250ka$LR04_Age_ka_mid,
           y = c(rep(3.0, 4),
                 seq(3.2, 2.7, length.out = 6),
                 rep(3.0, 2)),
           size = 3) +
  geom_line(data = lisiecki2005,                # add d18O line
            aes(Time,
                d18O)) +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()

## -----------------------------------------------------------------------------
ggplot() +
  geom_segment(data = mis_last_250ka, # add MIS lines
               aes(x =    LR04_Age_ka_end,
                   xend = LR04_Age_ka_start,
                   y =    seq(3, 2.5, length.out = nrow(mis_last_250ka)),
                   yend = seq(3, 2.5, length.out = nrow(mis_last_250ka))),
               colour = "blue",
               size = 1) +
  annotate("text",
           label = mis_last_250ka$label_MIS,
           x =     mis_last_250ka$LR04_Age_ka_mid,
           y = c(seq(2.9, 2.7, length.out = 4),
                 seq(3.1, 2.8, length.out = 5),
                 seq(2.4, 2.3, length.out = 3)),
           size = 3) +
  geom_line(data = lisiecki2005,                # add d18O line
            aes(Time,
                d18O)) +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()

## -----------------------------------------------------------------------------
ggplot() +
  annotate("rect",
           xmin = mis_last_250ka$LR04_Age_ka_end,
           xmax = mis_last_250ka$LR04_Age_ka_start,
           ymin = -Inf,
           ymax = Inf,
        alpha = .2,
        fill = rep(c("grey70", "white"),
                   nrow(mis_last_250ka)/2)) +
  annotate("text",
           label = mis_last_250ka$label_MIS,
           x =     mis_last_250ka$LR04_Age_ka_mid,
           y = c(rep(3, 4),
                 seq(3.1, 2.8, length.out = 5),
                 rep(3, 3)),
           size = 3) +
  geom_line(data = lisiecki2005,                # add d18O line
            aes(Time,
                d18O)) +
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_reverse(name = bquote(delta^18*O)) +
  theme_bw()

## -----------------------------------------------------------------------------
ggplot() +
  annotate("rect",
           xmin = mis_last_250ka$LR04_Age_ka_end ,
           xmax = mis_last_250ka$LR04_Age_ka_start,
           ymin = -Inf,
           ymax = Inf,
        alpha = .2,
        fill = rep(c("grey70", "white"),
                   nrow(mis_last_250ka)/2)) +
  annotate("text",
           label = mis_last_250ka$label_MIS,
           x =     mis_last_250ka$LR04_Age_ka_mid,
           y = c(rep(9, 4),
                 seq(8.1, 9.9, length.out = 5),
                 rep(9, 3)),
           size = 3) +
  # add in other geoms here...
  scale_x_continuous(limits = c(0, 250),
                     name = "x 1000 years ago") +
  scale_y_continuous(limits = c(0, 10),
                  name = "Some other variable (units)") +
  theme_bw()

